pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -554,19 +560,25 @@ func newAWSCloud(config io.Reader, awsServices AWSServices) (*AWSCloud, error) {
 	if len(zone) <= 1 {
 		return nil, fmt.Errorf("invalid AWS zone in config file: %s", zone)
 	}
-	regionName := zone[:len(zone)-1]
+
+	regionName := cfg.Global.Region
+	if len(regionName) <= 1 {
```

Comment:
Do you mean `regionName == ""` or `len(regionName) == 0`?  The len(zone) <=1 test above is because we're about to slice the last character off.

Maybe this logic is more correct?

```
regionName := cfg.Global.Region
  if regionName == "" {
    if len(zone) <= 1 {
      return nil, fmt.Errorf("invalid AWS zone in config file: %s", zone)
    }
    regionName = zone[:len(zone)-1]
}
```